src/languages/smtlib/parseSmtlib.mly

```diff
@@ -137,60 +170,109 @@ command_option:
 ;
 
 command:
-  | OPEN POP n=NUMERAL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.pop ~loc (int_of_string n) }
-  | OPEN PUSH n=NUMERAL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.push ~loc (int_of_string n) }
-
   | OPEN ASSERT t=term CLOSE
     { let loc = L.mk_pos $startpos $endpos in S.assert_ ~loc t }
   | OPEN CHECK_SAT CLOSE
     { let loc = L.mk_pos $startpos $endpos in S.check_sat ~loc () }
 
-  | OPEN SET_LOGIC s=SYMBOL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.set_logic ~loc s }
+  | OPEN DECLARE_CONST s=SYMBOL ty=sort CLOSE
+    { let id = I.(mk term s) in
+      let loc = L.mk_pos $startpos $endpos in
+      S.fun_decl ~loc id [] ty }
 
-  | OPEN GET_INFO i=KEYWORD CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.get_info ~loc i }
-  | OPEN SET_INFO c=command_option CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.set_info ~loc c }
+  | OPEN DECLARE_DATATYPE s=SYMBOL OPEN clist=datatype_dec CLOSE CLOSE
+    { let id = I.(mk sort s) in
+      let loc = L.mk_pos $startpos $endpos in
+      S.dtype_decl ~loc id [] (snd clist) }
 
-  | OPEN GET_OPTION k=KEYWORD CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.get_option ~loc k }
-  | OPEN SET_OPTION c=command_option CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.set_option ~loc c }
+  | OPEN DECLARE_DATATYPES OPEN sl=sort_dec+ CLOSE OPEN lcl=datatype_dec+ CLOSE CLOSE
+    { let loc = L.mk_pos $startpos $endpos in
+      let dl = List.map2 (fun sd cl -> S.dtype_decl ~loc (fst sd) (fst cl) (snd cl)) sl lcl 
+      in
+      S.data ~loc ~attrs:[] dl }
+
+  | OPEN DECLARE_FUN s=SYMBOL OPEN args=sort* CLOSE ty=sort CLOSE
+    { let id = I.(mk term s) in
+      let loc = L.mk_pos $startpos $endpos in
+      S.fun_decl ~loc id args ty }
 
   | OPEN DECLARE_SORT s=SYMBOL n=NUMERAL CLOSE
     { let id = I.(mk sort s) in
       let loc = L.mk_pos $startpos $endpos in
       S.type_decl ~loc id (int_of_string n) }
+
+  | OPEN DECLARE_HEAP sl=sort_pair+ CLOSE
+    { let loc = L.mk_pos $startpos $endpos in
+      let hid = I.(mk sort "heap_t") in
+      let hty = T.annot ~loc (T.const ~loc hid) sl in
+      S.type_def ~loc hid [] hty }
+    
+  | OPEN DEFINE_FUN s=SYMBOL OPEN args=sorted_var* CLOSE ret=sort body=term CLOSE
+    { let id = I.(mk term s) in
+      let loc = L.mk_pos $startpos $endpos in
+      S.fun_def ~loc id args ret body }
+
+  | OPEN DEFINE_FUN_REC s=SYMBOL OPEN args=sorted_var* CLOSE ret=sort body=term CLOSE
+    { let id = I.(mk term s) in
+      let loc = L.mk_pos $startpos $endpos in
+      let rty = T.annot ~loc ret [T.const ~loc I.(mk attr "rec")] in
```

Comment:
For this line, it would be better to use the location for `ret` rather than the location of the whole command. This is easily done usgin `$startpos(ret)` and `$endpos(ret)`